### PR TITLE
CMRARC-836: Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by htranho**

### CODE REVIEW

File: config/initializers/permissions_policy.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# information see https://developers.google.com/web/updates/2018/06/feature-policy
+#
+# Rails.application.config.permissions_policy do |f|
+#   f.camera      :none
+#   f.gyroscope   :none
+#   f.microphone  :none
+#   f.usb         :none
+#   f.fullscreen  :self
+#   f.payment     :self, "https://secure.example.com"
+# end
```
Comment (mandyparson) [inline config/initializers/permissions_policy.rb:11]: Is this file supposed to be commented out?

**Review by mandyparson — COMMENTED** [review]
yay!! My rake tests work now thanks to this

File: Gemfile [hunk 2]
```diff
@@ -2,14 +2,17 @@ source 'https://rubygems.org'
...
 # Use sqlite3 as the database for Active Record
-gem 'sqlite3', group: [:test]
+gem 'sqlite3', '1.7.3', group: [:test]
 # use pg for local and prod development
 gem 'pg', group: [:test, :development, :production]
 gem 'webpacker'
 gem 'react-rails'
 
+gem 'sprockets-rails'
+#gem 'rspec-core'
```
Comment (cgokey) [inline Gemfile:14]: remove

File: config/initializers/new_framework_defaults_7_0.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file eases your Rails 7.0 framework defaults upgrade.
+#
+# Uncomment each configuration one by one to switch to the new default.
+# Once your application is ready to run with all new defaults, you can remove
+# this file and set the `config.load_defaults` to `7.0`.
```
Comment (cgokey) [inline config/initializers/new_framework_defaults_7_0.rb:7]: I see you added `config.load_defaults` to 7.0, but I think the lines below should be uncommented one by one based on the instructions above.

File: Gemfile (hunk 2 above)
Comment (htranho) [reply Gemfile]: Done

File: app/helpers/record_helper.rb [hunk 4]
```diff
@@ -36,14 +36,15 @@ def is_number?(object)
   # checks if the granule can be associated
   # note if the checks fail, the caller should not associate the granule record to the collection.
   def can_associate_granule?(granule_record, collection_state)
-    # it is ok to associate granule records without checks if in open or in_arc_review
-    # May 7 2020: Added status 'closed', this is a temporary change to allow users to associate
-    # closed records until arc team either finishes reviewing all legacy imported records or
-    # we fix the bug that is preventing users from marking all colors for a record.
-    return [true, nil] if %w(open in_arc_review closed).include?(collection_state)
+    return [true, nil] if %w(open in_arc_review).include?(collection_state)
```
[the remark this answers is not in the document]
Comment (htranho) [reply app/helpers/record_helper.rb:39]: Done

File: config/initializers/permissions_policy.rb (hunk 1 above)
Comment (htranho) [reply config/initializers/permissions_policy.rb:11]: We don't use any of these.

File: config/initializers/new_framework_defaults_7_0.rb (hunk 3 above)
Comment (htranho) [reply config/initializers/new_framework_defaults_7_0.rb:7]: This file is no longer needed after config.load_defaults is set to 7.0. I keep this file only for documentation purpose.

**Review by cgokey — APPROVED** [review]

**Review by tle1989 — APPROVED** [review]

**Pull request merged**